### PR TITLE
add github action for mayhem for api

### PR DESCRIPTION
**Pull request by dlowe**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Add a github action to run [Mayhem for API](https://mayhem4api.forallsecure.com/) (which is a free automated testing tool for http APIs) against the api: given the openapi spec and api server, it generates and runs random test payloads, looking for exceptional behavior.

Disclosure: working on this tool is my day job!

As with any github workflow, test results can be consumed through github. Alternatively, Mayhem for API also has a dashboard, over here: https://mayhem4api.forallsecure.com/mayhemheroes/forem.

The currently checked-in spec only documents `GET /api/articles`, so that's all that's currently tested. Still, in a few seconds' testing, it finds an Internal Server Error that *probably* just boils down to data validation, but might be more serious.

To make it work in the upstream repo, there's a bit of out-of-PR effort to integrate:

- sign up for a free mapi account, creating a forem organization
- create a mapi service account within the forem organization
- modify the mapi.yaml action in this PR to point at your mapi organization
- add the service account's API token to github as a repository secret named MAPI_TOKEN

## Related Tickets & Documents

N/A

## QA Instructions, Screenshots, Recordings

This is strictly a build change.

### UI accessibility concerns?

N/A

## Added/updated tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

### CODE REVIEW

File: .github/workflows/mapi.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: mapi
+on:
+  pull_request:
+    branches: [main]
+  push:
+    branches: [main]
```
[the remark this answers is not in the document]
Comment (dlowe) [reply .github/workflows/mapi.yml]: Makes sense to me. Done in bd17a5c5f6a49a37babd61c64669e88e3fb41878 :)

File: .github/workflows/mapi.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 1
+
+      - name: Start the service
+        run: docker-compose up -d
```
[the remark this answers is not in the document]
Comment (dlowe) [reply .github/workflows/mapi.yml:17]: Well, it worked well enough to bring up the service and respond to mapi requests :)

**Comment by dlowe** [comment]
> Thank you for this contribution @dlowe. The product/service seems quite interesting, but we're currently in the process of updating our API + docs, including the OpenAPI spec file, so there's a chance we'll have to wait a few weeks before we can consider merging this PR.

Thanks for reviewing, and my pleasure :)

Obviously I'm biased, but if you're actively working on the openapi spec, I think mapi is a great tool for ensuring an accurate spec! As a CLI dev tool (outside CI) it's free and (hopefully!) easy to use, and one of the default outputs is warnings about when the spec doesn't match the empirical behavior of the service.

**Comment by maestromac** [comment]
Since this is a bit premature for us and we are still undecided, we are going to close this for time being. Regardless, we really appreciate your contribution @dlowe .

**Comment by dlowe** [comment]
> Since this is a bit premature for us and we are still undecided, we are going to close this for time being. Regardless, we really appreciate your contribution @dlowe .

Makes sense to me, thanks for letting me know!

**Pull request closed without merging**